components/system-config-mgt/org.wso2.carbon.identity.system.config.mgt/src/main/java/org/wso2/carbon/identity/system/config/mgt/advisory/DBBasedAdminBannerDAO.java

```diff
@@ -0,0 +1,178 @@
+import org.wso2.carbon.identity.configuration.mgt.core.model.Attribute;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resource;
+import org.wso2.carbon.identity.configuration.mgt.core.model.ResourceFile;
+import org.wso2.carbon.identity.system.config.mgt.internal.SystemConfigMgtServiceHolder;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This class is used to manage storage of the Admin Advisory Banner configurations in the configuration-mgt store.
+ */
+public class DBBasedAdminBannerDAO implements AdminAdvisoryBannerDAO {
+
+    public static final String ADVISORY_BANNER_RESOURCE_TYPE = "ADMIN_ADVISORY_BANNER";
+    public static final String ADVISORY_BANNER_RESOURCE_NAME = "ADMIN_ADVISORY_BANNER_RESOURCE";
+    public static final String ENABLE_BANNER = "enableBanner";
+    public static final String BANNER_CONTENT = "bannerContent";
+    public static final String RESOURCE_NOT_EXISTS_ERROR_CODE = "CONFIGM_00017";
+
+    private static final Log LOG = LogFactory.getLog(DBBasedAdminBannerDAO.class);
+
+    @Override
+    public void saveAdminAdvisoryConfig(AdminAdvisoryBannerDTO adminAdvisoryBannerDTO, String tenantDomain)
+            throws AdminAdvisoryMgtException {
+
+        Resource resource = buildResourceFromAdvisoryBannerDTO(adminAdvisoryBannerDTO);
+        try {
+            if (isAdminAdvisoryResourceExists(ADVISORY_BANNER_RESOURCE_TYPE, ADVISORY_BANNER_RESOURCE_NAME)) {
+                getConfigurationManager().replaceResource(ADVISORY_BANNER_RESOURCE_TYPE, resource);
+            } else {
+                getConfigurationManager().addResource(ADVISORY_BANNER_RESOURCE_TYPE, resource);
+            }
+        } catch (ConfigurationManagementException e) {
+            throw new AdminAdvisoryMgtException("Error occurred while saving advisory banner configuration.", e);
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Admin advisory banner configuration saved successfully in configuration-store for tenant: "
+                    + tenantDomain);
+        }
+    }
+
+    @Override
+    public Optional<AdminAdvisoryBannerDTO> loadAdminAdvisoryConfig(String tenantDomain)
+            throws AdminAdvisoryMgtException {
+
+        try {
+            Resource resource = getConfigurationManager().getResource(ADVISORY_BANNER_RESOURCE_TYPE,
+                    ADVISORY_BANNER_RESOURCE_NAME);
+            if (resource == null) {
+                return Optional.empty();
+            }
+
+            List<ResourceFile> resourceFiles = getConfigurationManager().getFiles(ADVISORY_BANNER_RESOURCE_TYPE,
+                    ADVISORY_BANNER_RESOURCE_NAME);
+
+            if (resourceFiles.isEmpty() || StringUtils.isBlank(resourceFiles.get(0).getId())) {
```

Comment:
Lets make a local veriable with this and`resourceFiles.get(0)` lets check for null here.

Reply:
Addressed with: https://github.com/wso2/carbon-identity-framework/compare/1592056d8c0cdd688460da3409c96537ba73326c..c87d5bd1af45d97e6f34ba9ba9ef0ea25e0250de